### PR TITLE
CLDR-16467 update CLDR test data

### PR DESCRIPTION
**Pull request by macchiati**
CLDR-16467

This took longer than expected. Because we didn't have a BRS item some of these were dated.

The instructions are (temporarily) in https://docs.google.com/document/d/1-RC99npKcSSwUoYGkSzxaKOe76gYRkWhGdFzCdIBCu4/edit#

common/properties/coverageLevels.txt 
- Just copyright

common/testData/localeIdentifiers/localeCanonicalization.txt
- Updates to for additional language codes

common/testData/localeIdentifiers/localeDisplayName.txt
- Additions because the last time this was generated was a while ago

 common/testData/personNameTest/en_IN.txt
- minor changes from data updates

common/testData/units/unitsTest.txt
- mostly new units. Some conversion corrections for previous versions.

tools/cldr-code/src/main/java/org/unicode/cldr/tool/GenerateLocaleIDTestData.java
- The code for localeDisplayName.txt was missing, so recreated.

tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestUnits.java
- The process was to copy and paste from the console, so made some basic changes to write the files directly instead.

- [ ] This PR completes the ticket.

ALLOW_MANY_COMMITS=true

### CODE REVIEW

**Review by pedberg-icu — APPROVED** [review]

File: common/testData/localeIdentifiers/localeCanonicalization.txt [hunk 1]
```diff
@@ -35,8 +35,8 @@ hy_arevela	;	hy
 hy_arevmda	;	hyw
 hy_arevmda_arevela	;	hyw
 hye_arevmda	;	hyw
-no_bokmal_nynorsk	;	nb
-no_nynorsk_bokmal	;	nb
+no_bokmal_nynorsk	;	no
+no_nynorsk_bokmal	;	no
```
Comment (pedberg-icu) [inline common/testData/localeIdentifiers/localeCanonicalization.txt:39]: @macchiati I think this change in lines 38-39 is wrong (and so does the code in ICU4C, ICU4J). I filed https://unicode-org.atlassian.net/browse/CLDR-16476 about this.

**Comment by macchiati** [comment]
I'm taking that ticket and have a fix.

File: common/testData/localeIdentifiers/localeCanonicalization.txt [hunk 2]
```diff
@@ -1153,6 +1221,7 @@ aaa_Adlm_AC_arevela_fonipa	;	aaa_Adlm_AC_fonipa
 aaa_Adlm_AC_arevmda_fonipa	;	aaa_Adlm_AC_fonipa
 aaa_Adlm_AC_bokmal_fonipa	;	aaa_Adlm_AC_fonipa
 aaa_Adlm_AC_fonipa_hakka	;	aaa_Adlm_AC_fonipa
+aaa_Adlm_AC_fonipa_hepburn_heploc	;	aaa_Adlm_AC_alalc97_fonipa
```
Comment (kipcole9) [inline common/testData/localeIdentifiers/localeCanonicalization.txt:1224]: Should this canonicalise to aaa-Adlm-AC-alalc97-fonipa-hepburn ?

1. Alias "heploc" => "alalc97"
2. No aliases for "fonipa" or "hepburn"
3. Variant sort order is lexical: "alalc97-fonipa-hepburn"

**Comment by macchiati** [comment]
Thanks, Kip

I'll check that out.

**Comment by kipcole9** [comment]
Mark, It was not a test data error. I deleted the comment.  Nothing to see here.

**Comment by macchiati** [comment]
Ok, thanks for letting me know

**Pull request merged**